File: frontend/src/api/category.js
import request from './request'

/**
 * 获取所有分类（树形结构）
 * GET /category/tree
 * 返回: List<Category> 树形列表
 *   Category { id, name, parentId, sort, status, children: [] }
 */
export function getCategoryTree() {
  return request.get('/category/tree')
}

/**
 * 获取所有分类（扁平列表）
 * GET /category/list
 * 返回: List<Category>
 *   Category { id, name, parentId, sort, status }
 */
export function getCategoryList() {
  return request.get('/category/list')
}

File: backend/src/main/java/com/moying/project_test/entity/Category.java
package com.moying.project_test.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * <p>
 * 商品分类表
 * </p>
 *
 * @author moying
 * @since 2026-08-26
 */
@Getter
@Setter
@ToString
@TableName("t_category")
public class Category implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 分类ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 分类名称
     */
    @TableField("name")
    private String name;

    /**
     * 父分类ID, 0为一级分类
     */
    @TableField("parent_id")
    private Long parentId;

    /**
     * 排序号（越小越靠前）
     */
    @TableField("sort")
    private Integer sort;

    /**
     * 状态: 0-禁用 1-启用
     */
    @TableField("status")
    private Byte status;

    /**
     * 创建时间
     */
    @TableField("create_time")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField("update_time")
    private LocalDateTime updateTime;
}
